core/modules/catalog-magento/store/category/actions.ts

```diff
@@ -0,0 +1,132 @@
+// import Vue from 'vue'
+import { ActionTree } from 'vuex'
+import * as types from './mutation-types'
+import RootState from '@vue-storefront/core/types/RootState'
+import CategoryState from './CategoryState'
+import { quickSearchByQuery } from '@vue-storefront/core/lib/search'
+import { buildFilterProductsQuery } from '@vue-storefront/core/helpers'
+import config from 'config'
+import trim from 'lodash-es/trim'
+import toString from 'lodash-es/toString'
+import { currentStoreView } from '@vue-storefront/core/lib/multistore'
+import { optionLabel } from '../../helpers/optionLabel'
+import { router } from '@vue-storefront/core/app'
+import FilterVariant from '../../types/FilterVariant';
+
+const actions: ActionTree<CategoryState, RootState> = {
+  /**
+   * Initialise category module.
+   * - gets available categories
+   * - gets available filters for current category
+   */
+  async initCategoryModule ({ getters, dispatch }) {
+    if (!getters.getCategories.length) {
+      await dispatch('getCategories')
+      await dispatch('getAvailableFilters')
+    }
+  },
+  async searchProducts ({ commit, getters, dispatch }, { filters, route } = {}) {
+    await dispatch('initCategoryModule')
+    const searchQuery = getters.calculateFilters(filters)
+    const searchCategory = getters.calculateCurrentCategory(route)
+    let filterQr = buildFilterProductsQuery(searchCategory, searchQuery.filters)
+    const searchResult = await quickSearchByQuery({ query: filterQr, sort: searchQuery.sort })
+    commit(types.CATEGORY_SET_PRODUCTS, searchResult.items)
+
+    return searchResult.items
+  },
+  async getCurrentCategory ({ dispatch, rootState }) {
+    return await dispatch('category/single', { key: config.products.useMagentoUrlKeys ? 'url_key' : 'slug', value: rootState.route.params.slug }, {root: true})
+  },
+  async fetchCategories ({ dispatch }) {
+    const res = await dispatch('category/list', {}, {root: true})
+    return res.items
+  },
+  async getCategories ({ commit, dispatch }) {
+    const categories = await dispatch('fetchCategories')
+    commit(types.CATEGORY_SET_CATEGORIES, categories)
+    return categories
+  },
```

Comment:
sure, we could explain it better in documentation. In my assumption:
fetch - invokes api, gets data from server and returning them - without mutating state or reading currently cached in state results
get - does not directly call api, it can read data from store and return them or invoke fetch when there is no data (or fetch either way), then store result in state.

That's why when somebody needs to just get something from API and does not want to mutate current state (for example he needs data in another module or just for some preview) would just invoke fetch.
Of course naming can be changed, but i'd keep that level of abstraction.

Reply:
sure, we could explain it better in documentation. In my assumption:
fetch - invokes api, gets data from server and returning them - without mutating state or reading currently cached in state results
get - does not directly call api, it can read data from store and return them or invoke fetch when there is no data (or fetch either way), then store result in state.

That's why when somebody needs to just get something from API and does not want to mutate current state (for example he needs data in another module or just for some preview) would just invoke fetch.
Of course naming can be changed, but i'd keep that level of abstraction.